# By ZDBioHazard, Jan 17, 2012
#
# This patch reformats the play controls on the main window into a single
# table rather than a vbox of 2 separate tables. There is no functional
# difference, I just feel it is cleaner this way. Saves 9 lines. ;)
#
--- quodlibet/qltk/controls.py
+++ quodlibet/qltk/controls.py
@@ -266,40 +266,31 @@
         return self.__item.set_active(active)
     active = property(__get_active, __set_active)
 
-class PlayControls(gtk.VBox):
+class PlayControls(gtk.Table):
     def __init__(self, player, library):
-        super(PlayControls, self).__init__(spacing=3)
-
-        upper = gtk.Table(rows=1, columns=3, homogeneous=True)
-        upper.set_row_spacings(3)
-        upper.set_col_spacings(3)
+        super(PlayControls, self).__init__(rows=2, columns=3, homogeneous=True)
+        self.set_row_spacings(3)
+        self.set_col_spacings(3)
 
         prev = gtk.Button()
         prev.add(gtk.image_new_from_stock(gtk.STOCK_MEDIA_PREVIOUS, SIZE))
-        upper.attach(prev, 0, 1, 0, 1)
+        self.attach(prev, 0, 1, 0, 1)
 
         play = gtk.ToggleButton()
         play.add(gtk.image_new_from_stock(gtk.STOCK_MEDIA_PLAY, SIZE))
-        upper.attach(play, 1, 2, 0, 1)
+        self.attach(play, 1, 2, 0, 1)
 
         safter = StopAfterMenu(player)
 
         next = gtk.Button()
         next.add(gtk.image_new_from_stock(gtk.STOCK_MEDIA_NEXT, SIZE))
-        upper.attach(next, 2, 3, 0, 1)
-
-        lower = gtk.Table(rows=1, columns=3, homogeneous=True)
-        lower.set_row_spacings(3)
-        lower.set_col_spacings(3)
+        self.attach(next, 2, 3, 0, 1)
 
         self.volume = Volume(player)
-        lower.attach(self.volume, 0, 1, 0, 1)
+        self.attach(self.volume, 0, 1, 1, 2)
 
         seekbar = SeekBar(player, library)
-        lower.attach(seekbar, 1, 3, 0, 1)
-
-        self.pack_start(upper, expand=False)
-        self.pack_start(lower, expand=False)
+        self.attach(seekbar, 1, 3, 1, 2)
 
         prev.connect_object('clicked', self.__previous, player)
         play.connect('toggled', self.__playpause, player)
